modules/core/test/src/smithy4s/ReservedTypeSmokeSpec.scala

```diff
@@ -0,0 +1,37 @@
+/*
+ *  Copyright 2021-2022 Disney Streaming
+ *
+ *  Licensed under the Tomorrow Open Source Technology License, Version 1.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *     https://disneystreaming.github.io/TOST-1.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package smithy4s
+
+import munit._
+import smithy4s.example.collision.ReservedNameService
+
+class ReservedTypeSmokeSpec() extends FunSuite {
+
+  test("Idiomatic ADTs can be generated from unions") {
```

Comment:
what exactly is being tested here?

Reply:
ahh I guess the test name is garbage, but we're just ensuring that `ReservedNameService` is available (because codegen & compilation was successful (so was collision handling))

Reply:
Yeah I copied another spec and forgot to rename the test. 

We're testing that the generated code actually uses types from the Scala stdlib, and that the methods are not renamed with underscores or whatnot.